src/v3/src/util/formUtils.ts

```diff
@@ -390,3 +392,90 @@ export const shouldHideIdentifier = (
 
   return false;
 };
+
+const getPhoneVerificationSubtitleTextContent = (
+  step: string,
+  primaryMethod: string,
+  phoneNumber?: string,
+  nickname?: string,
+): string => {
+  if (![
+    IDX_STEP.CHALLENGE_AUTHENTICATOR,
+    IDX_STEP.AUTHENTICATOR_VERIFICATION_DATA,
+  ].includes(step)
+    || !['sms', 'voice'].includes(primaryMethod)) {
+    // If step or method is invalid, return.
+    return '';
+  }
+  const i18nMap: Record<string, Record<string, Record<string, string>>> = {
```

Comment:
```suggestion
  const i18nMap: Record<IDX_STEP.CHALLENGE_AUTHENTICATOR | IDX_STEP.AUTHENTICATOR_VERIFICATION_DATA, Record<'sms' | 'voice', Record<'withPhoneWithNickName' | 'withPhoneWithoutNickName' | 'withoutPhone', string>>> = {
```
some formatting or defining some local union types might make it cleaner but you get the idea